### PR TITLE
PhishlabsIOC_EIR integration

### PR DESCRIPTION
**Pull request by GalRabin**
## Status
Ready

## Related Issues
https://github.com/demisto/etc/issues/15359

## Description
New EIR integration of PhishLabs company.

## Required version of Demisto
x.x.x

## Does it break backward compatibility?
   - No

## Must have
- [x] Tests
- [x] Documentation (with link to it)
- [x] Code Review

## Technical writer review
Mention and link to the files that require a technical writer review.
- [x] [YAML file](https://github.com/demisto/content/blob/phishlabs-eir/Integrations/PhishLabsIOC_EIR/PhishLabsIOC_EIR.yml)
- [x] [README file](https://github.com/demisto/content/blob/403bcdadf4e28c95748fc70dfa4f8369d0c91f0d/Integrations/PhishLabsIOC_EIR/README.md)
- [x] [Description.md](https://github.com/demisto/content/blob/phishlabs-eir/Integrations/PhishLabsIOC_EIR/PhishLabsIOC_EIR_description.md)

### CODE REVIEW

File: Integrations/PhishLabsIOC_EIR/PhishLabsIOC_EIR.yml [hunk 1]
```diff
@@ -0,0 +1,398 @@
...
+  type: 0
+- display: User
+  name: user
+  required: true
+  type: 0
+- display: Password
+  name: password
+  required: true
+  type: 4
+- display: Fectches incidents
```
Comment (Itay4) [inline Integrations/PhishLabsIOC_EIR/PhishLabsIOC_EIR.yml]: fetch params are wrong, please make sure you have them all and named correctly

File: Integrations/PhishLabsIOC_EIR/PhishLabsIOC_EIR.yml [hunk 2]
```diff
@@ -0,0 +1,398 @@
...
+- display: Use system proxy settings
+  name: proxy
+  required: false
+  type: 8
+- defaultvalue: 1 hours
+  display: First fetch timestamp (<number> <time unit>, e.g., 12 hours, 7 days)
+  name: fetch_time
+  required: false
+  type: 0
+description: Get live feeds of IOC data from PhishLabs.
```
Comment (yaakovi) [inline Integrations/PhishLabsIOC_EIR/PhishLabsIOC_EIR.yml]: move another fetch parameter to limit the number of incident created by a single fetch-incident iteration

File: Integrations/PhishLabsIOC_EIR/PhishLabsIOC_EIR_description.md [hunk 3]
```diff
@@ -0,0 +1,2 @@
+ Get Email Incidents report from PhishLabs IOC service.\
```
Comment (yaakovi) [inline Integrations/PhishLabsIOC_EIR/PhishLabsIOC_EIR_description.md]: remove leading spaces in both lines

File: TestPlaybooks/playbook-PhishlabsIOC_EIR-Test.yml [hunk 4]
```diff
@@ -0,0 +1,349 @@
+id: PhishlabsIOC_EIR-Test
```
Comment (yaakovi) [inline TestPlaybooks/playbook-PhishlabsIOC_EIR-Test.yml]: add a task for `FetchFromInstance` in order to test the fetch-incident mechanism

File: Integrations/PhishLabsIOC_EIR/PhishLabsIOC_EIR.py [hunk 5]
```diff
@@ -0,0 +1,432 @@
...
+            Response JSON as dictionary
+        """
+        suffix = "/incidents/EIR"
+        return self._http_request('GET',
+                                  url_suffix=suffix,
+                                  params={**kwargs,
+                                          'offset': offset,
+                                          'limit': limit})
+
+    def phishlabs_ioc_eir_get_incident_by_id(self, **kwargs) -> Dict:
```
Comment (yaakovi) [inline Integrations/PhishLabsIOC_EIR/PhishLabsIOC_EIR.py]: no actual reason to use kwargs here.
you have only one acceptable argument and it is mandatory

File: Integrations/PhishLabsIOC_EIR/PhishLabsIOC_EIR.py [hunk 6]
```diff
@@ -0,0 +1,432 @@
...
+
+    Returns:
+        'ok' if test successful.
+
+    Raises:
+        DemistoException: If test failed.
+    """
+    results = client.test_module()
+    if 'incidents' in results:
+        return None, None, 'ok'
```
[the remark this answers is not in the document]
Comment (GalRabin) [reply Integrations/PhishLabsIOC_EIR/PhishLabsIOC_EIR.py]: yes I am sure, checked it

File: Integrations/PhishLabsIOC_EIR/PhishLabsIOC_EIR.py [hunk 7]
```diff
@@ -271,55 +288,76 @@ def test_module_command(client: Client, *_) -> Tuple[None, None, str]:
...
-    raw_response = client.phishlabs_ioc_eir_get_incidents(**kwargs)
-    while raw_response.get('metadata', {}).get('count'):
+        datetime_new_last_run = last_run
+    # Query incidents by limit and creation time
+    total = 0
+    offset = 50
+    limit_incidents = int(limit)
+    limit_page = 50
+    if limit_incidents <= 50:
+        limit_page = limit_incidents
```
Comment (yaakovi) [inline Integrations/PhishLabsIOC_EIR/PhishLabsIOC_EIR.py]: `limit_page = min(limit_incidents, 50)`

File: Integrations/PhishLabsIOC_EIR/PhishLabsIOC_EIR.py [hunk 8]
```diff
@@ -271,55 +288,76 @@ def test_module_command(client: Client, *_) -> Tuple[None, None, str]:
...
-        raw_response = client.phishlabs_ioc_eir_get_incidents(offset=offset, **kwargs)
-    if incidents:
-        for incident in incidents:
+        incidents_raw += raw_response.get('incidents', [])
+        total += int(raw_response.get('metadata', {}).get('count'))
+        if total >= limit_incidents:
+            break
+        if limit_incidents - total < 50:
+            limit_page = limit_incidents - total
+        offset += limit_page
```
Comment (yaakovi) [inline Integrations/PhishLabsIOC_EIR/PhishLabsIOC_EIR.py]: shouldn't offset be updated with the length of the last fetched data (`raw_response.get('metadata', {}).get('count')`)?

File: Integrations/PhishLabsIOC_EIR/PhishLabsIOC_EIR.yml [hunk 9]
```diff
@@ -104,7 +105,7 @@ script:
       secret: false
     - default: false
       defaultValue: '50'
-      description: Limit amounts of incdients (0-50, default 50)
+      description: Limit amounts of incidents  (0-50, default 25)
```
Comment (yaakovi) [inline Integrations/PhishLabsIOC_EIR/PhishLabsIOC_EIR.yml]: default is not 25, update the default value or the description.

File: Integrations/PhishLabsIOC_EIR/PhishLabsIOC_EIR.py (hunk 8 above)
Comment (GalRabin) [reply Integrations/PhishLabsIOC_EIR/PhishLabsIOC_EIR.py]: You right it, it can be a problem if I don't get the expected amount.

**Comment by GalRabin** [comment]
@kirbles19 
I will appreciate your docs approval (Customer require this integration ASPA):
- [ ] [YAML file](https://github.com/demisto/content/blob/phishlabs-eir/Integrations/PhishLabsIOC_EIR/PhishLabsIOC_EIR.yml)
- [ ] [README file](https://github.com/demisto/content/blob/403bcdadf4e28c95748fc70dfa4f8369d0c91f0d/Integrations/PhishLabsIOC_EIR/README.md)
- [ ] [Description.md](https://github.com/demisto/content/blob/phishlabs-eir/Integrations/PhishLabsIOC_EIR/PhishLabsIOC_EIR_description.md)

Thanks

**Comment by kirbles19** [comment]
@GalRabin Done.

**Pull request merged**